cpp/src/io/parquet/reader_impl_helpers.cpp

```diff
@@ -1712,6 +1712,11 @@ aggregate_reader_metadata::select_columns(
           });
         if (found_path != all_paths.end()) {
           valid_selected_paths.push_back({selected_path, found_path->schema_idx});
+        } else {
+          // Ensure that selected path matches a path in all_paths
+          CUDF_EXPECTS(found_path != all_paths.end(),
+                       "Encountered non existent column in selected path",
```

Comment:
```suggestion
                       "Encountered non-existent column in selected path",
```

Also, could we report the column we didn't find? That's likely a less frustrating experience for the user if they typoed something.